### PR TITLE
Add signed tree head to inclusion proof

### PR DESCRIPTION
**Pull request by asraa**
Part of https://github.com/sigstore/rekor/issues/257

Adds a signed tree head to an inclusion proof so that a user can use the InclusionProof to verify that an entry exists in the transparency log as long as they trust Rekor's key to trust the root hash.

I implemented this PR off of https://github.com/sigstore/rekor/pull/250 to avoid re-using `GetLeavesByHash`

### CODE REVIEW

File: openapi.yaml [hunk 1]
```diff
@@ -409,11 +390,32 @@ definitions:
           type: string
           description: SHA256 hash value expressed in hexadecimal format
           pattern: '^[0-9a-fA-F]{64}$'
+      signedTreeHead:
```
Comment (dlorenc) [inline openapi.yaml:396]: Can we reuse the signedTreeHead definition from line 329?

**Review by bobcallaway — APPROVED** [review]
LGTM!

**Comment by asraa** [comment]
Thanks! I'll merge in main but just FYI I'm looking into replacing this with a "signed promise" like SCT for ctlog that can be used to replace the need for copying the whole inclusion proof when verifying without calling rekor.
@priyawadhwa

Edit: Also in flux after the recent trillian change.

**Comment by asraa** [comment]
Closed with https://github.com/sigstore/rekor/pull/291

**Pull request closed without merging**